rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,211 @@
+
+In the future, Glides will move to importing scheduled trips from the [Transit Operational Data Standard (TODS)](https://ods.calitp.org/) file produced by Transit Data. This will similarly use trip IDs sourced from HASTUS and so the trip ID logic will not have to change, although the `schedule_id` we currently use will likely need to be replaced with a `service_id` to match the GTFS-based TODS standard. This change should be largely orthogonal to other changes proposed in this document, so work on TODS does not need to block adoption of these recommendations, or vice versa.
+
+The use of TODS will slightly modify the overall architecture diagram from above (assuming that RTR reads data from TODS as well):
+
+```mermaid
+  flowchart BT
+    hastus["HASTUS Schedule"]
+	tods["TODS"]
+	glides["Glides"]
+	rtr["RTR"]
+	ocs["OCS via Trike"]
+	predictions["TripUpdates"]
+	vehicles["VehiclePositions"]
+	hastus -- Trips --> tods
+	tods -- Trips --> glides
+	tods -- Trips --> rtr
+	glides -. Trip Assignments .-> rtr
+	glides -. Trainsheet Edits .-> rtr
+	ocs -. Vehicle Locations .-> rtr
+	rtr -. Vehicle Locations .-> glides
+	rtr -. Publishes .-> predictions
+	rtr -. Publishes .-> vehicles
+```
+
+## Realtime data: Glides trainsheets and RTR
+
+In the first stage, RTR and Glides will continue to maintain their own separate trip assignment systems, but Glides will begin sending HASTUS-based trip IDs in trainsheet update events. For scheduled trips, the [trip key published in `glides.trips_updated` events](https://mbta.github.io/schemas/events/glides/com.mbta.ctd.glides.trips_updated.v1#tripkey) will need to be updated in a new version of the schema to include this ID. The `serviceDate` will still need to be present in order to map from HASTUS schedule IDs to GTFS service IDs. The [Glides code that generates the trip key for the event](https://github.com/mbta/glides/blob/fbe9578c8400b955304c74d34e672460a0d17a23/lib/glides_web/channels/trainsheet_channel.ex#L1324) already has access to the `ScheduledTrip` with its trip ID, so once we are importing the trip ID from HASTUS we will simply be able to reference that. This will be a backwards-incompatible change, requiring a new version of the schema.
```

Comment:
@skyqrose and I had discussed this previously and personally I'm open minded to either approach. I think partially this comes down to what Transit Data thinks as the consumer (cc @bklebe since you've already been taking a look).

Reply:
It was @skyqrose who proposed the idea, I just tagged @bklebe for a Transit Data perspective. In any event it sounds like there's an emerging preference for the backwards-compatible approach, I'll just wait for someone from Transit Data to ultimately confirm before changing it.

Reply:
My soft preference is backward compatible as well. While we aren't producing predictions based on glides in production, we are parsing glides messages, so a breaking change means we have to coordinate rollouts.